examples/two-chip.rs

```diff
@@ -0,0 +1,723 @@
+                let cell = region.assign_advice(
+                    || "private input",
+                    config.advice[0],
+                    0,
+                    || value.ok_or(Error::SynthesisError),
+                )?;
+                num = Some(Number { cell, value });
+                Ok(())
+            },
+        )?;
+        Ok(num.unwrap())
+    }
+
+    /// Returns `d = (a + b) * c`.
+    fn add_and_mul(
+        &self,
+        layouter: &mut impl Layouter<F>,
+        a: <Self as FieldInstructions<F>>::Num,
+        b: <Self as FieldInstructions<F>>::Num,
+        c: <Self as FieldInstructions<F>>::Num,
+    ) -> Result<<Self as FieldInstructions<F>>::Num, Error> {
+        let ab = self.add(layouter.namespace(|| "a + b"), a, b)?;
+        self.mul(layouter.namespace(|| "(a + b) * c"), ab, c)
+    }
+
+    fn expose_public(
+        &self,
+        mut layouter: impl Layouter<F>,
+        num: <Self as FieldInstructions<F>>::Num,
+    ) -> Result<(), Error> {
+        let config = match self.config() {
+            FieldConfigEnum::Config(config) => config.clone(),
+            _ => unreachable!(),
+        };
+
+        layouter.assign_region(
+            || "expose public",
+            |mut region: Region<'_, F>| {
+                // Enable the public-input gate.
+                config.s_pub.enable(&mut region, 0)?;
+
+                // Load the output into the correct advice column.
+                let out = region.assign_advice(
+                    || "public advice",
+                    config.advice[1],
+                    0,
+                    || num.value.ok_or(Error::SynthesisError),
+                )?;
+                region.constrain_equal(&config.perm, num.cell, out)?;
+
+                // We don't assign to the instance column inside the circuit;
+                // the mapping of public inputs to cells is provided to the prover.
+                Ok(())
+            },
+        )
+    }
+}
+// ANCHOR_END: field-instructions-impl
+
+// ANCHOR: circuit
+/// The full circuit implementation.
+///
+/// In this struct we store the private input variables. We use `Option<F>` because
+/// they won't have any value during key generation. During proving, if any of these
+/// were `None` we would get an error.
+struct MyCircuit<F: FieldExt> {
+    a: Option<F>,
+    b: Option<F>,
+    c: Option<F>,
+}
+
+impl<F: FieldExt> Circuit<F> for MyCircuit<F> {
+    // Since we are using a single chip for everything, we can just reuse its config.
+    type Config = FieldConfigEnum;
+
+    fn configure(meta: &mut ConstraintSystem<F>) -> Self::Config {
+        // We create the two advice columns that FieldChip uses for I/O.
+        let advice = [meta.advice_column(), meta.advice_column()];
+
+        // We also need an instance column to store public inputs.
+        let instance = meta.instance_column();
+
+        let mut field_chip = FieldChip::new();
+        field_chip.configure(meta, advice, instance)
```

Comment:
```suggestion
        FieldChip::configure(meta, advice, instance)
```
(and associated changes here, similar to `simple-example.rs`)